ldap_validator/ldap_validator.py

```diff
@@ -0,0 +1,474 @@
+            connection.unbind()
+            return False
+    except LDAPException as e:
+        ldap_logger.exception("Failed connect to Server: " + str(server) + " with error " + str(e))
+        return False
+    # parse connection result, response, entries
+
+    parse_result: bool = parse_connection_results(ldap_logger, response, connection.result, connection.entries, verify)
+    connection.unbind()
+    return parse_result
+
+
+def parse_connection_results(ldap_logger, response, result, entries, verify):
+    """    Parse the results returned by connection to the defined LDAP server.\n
+
+    Argument:
+        response       - results returned in json format
+        result(dict)   - result on the connection created
+        entries(list)  - entries on the connection created
+        verify(bool) - Assert that a valid result is returned
+    Returns:
+        success flag(bool) - Success/Failure of query
+    """
+    results_returned = False
+    ldap_logger.debug("json response LDAP Server connection {}".format(str(response)))
+    ldap_logger.debug("connection result{}".format(pprint.pformat(result)))
+    ldap_logger.debug("connection entries {}".format(pprint.pformat(entries)))
+    if not(result):
+        ldap_logger.error("connection.result is empty")
+        return False
+    query_rc = int(result['result'])
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Search query return code: " + str(query_rc))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Response: " + str(response['entries']))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Result: " + str(result))
+    ldap_logger.debug("--------------------------------------------------------------------")
+    ldap_logger.debug("Entries: ")
+    for entry in entries:
+        results_returned = True
+        ldap_logger.debug(str(entry))
+    ldap_logger.debug("--------------------------------------------------------------------")
+
+    if (verify):
+        try:
+            assert(results_returned is True)
+        except AssertionError:
+            ldap_logger.exception("Error: LDAP Search did not return any connection entries")
+            return False
+    try:
+        assert(query_rc == 0 or query_rc == 4)
+    except AssertionError:
+        ldap_logger.exception("Error: LDAP Search query failed with return code: " + str(query_rc))
+        return False
+
+    ldap_logger.info("LDAP search queries completed successfully")
+    return True
+
+
+# --------------------------------------------------------------------------------------------------
+###################
+#     usage()     #
+###################
+def usage(exit_code: int):
+    """
+    Prints the usage information for the pre-install-report command and exits the program with the given exit_code.
```

Comment:
Remove reference to "pre-install-report"

Reply:
Upcoming push will have this text changed to "ldap-validator"